### ISSUE TITLE
v1.9.0 Broke ActionDispatch::Session::RedisStore

### ISSUE THREAD

**Issue by Lianowar** [redis-store/redis-store#336]
I found bug at v1.9.0. After update to this version I can't use redis_store as session_store in rails 6 api_only application. 
This [commit](https://github.com/redis-store/redis-store/commit/65d0133c641381e3c7201b657c586300c51f26ae)

```ruby  
  def set(key, value, options = nil)
        super(key, value, options || {})
        if options && REDIS_SET_OPTIONS.any? { |k| options.key?(k) }
          kwargs = REDIS_SET_OPTIONS.each_with_object({}) { |key, h| h[key] = options[key] if options.key?(key) }
          super(key, value, **kwargs)
        else
          super(key, value)
        end
      end
   end
```

At `options` there passed ```ActionDispatch::Request::Session::Options ``` and this class has not ``` key? ``` method and an exception occured

I fixed it by downgrade redis-store and redis to previous versions

**Comment by tubbo** [comment]
Did you also upgrade to Redis v4.2.x? I think you need v4.2 in order to use the latest Redis::Store.

**Comment by edmondchui** [comment]
I also have the same issue. I've tried Redis v5.0.8 and Redis 6.0.5 and both have the same issue.

**Comment by Lianowar** [comment]
@tubbo Yeah, I also have redis 4.2.x and to fix this issue downgrade it too

**Comment by leevlad** [comment]
Same issue here, this is preventing me from upgrading to sidekiq 6.1.0 (latest release at time of writing this).

**Comment by tubbo** [comment]
hey folks, sorry for the delay...I'm working on this now.

**Comment by tubbo** [comment]
I'll merge this to `master` once tests pass, then if y'all wouldn't mind seeing if it works in the real world, that would be great! I don't use redis sessions for any of my projects so I can't canary deploy it anywhere this time.

**Comment by Lianowar** [comment]
Yay! I will try it after your tests:)
Thank you)

**Comment by Lianowar** [comment]
@tubbo You do not remove that line
```
if options && REDIS_SET_OPTIONS.any? { |k| options.key?(k) }
```
but this line exactly broke all:))

**Comment by tubbo** [comment]
@Lianowar master branch on redis-rack should fix this. I'll release sometime today. https://github.com/redis-store/redis-rack/commit/57464f91af4d470de3dc3886f1aa362b0ed6d377

**Comment by edmondchui** [comment]
Hi @tubbo, how does the release of redis-rack solve this issue? And the latest release of redis-rack is still 2.1.2, not your version.

**Comment by brendon** [comment]
@edmondchui, @tubbo is converting the options to a hash before passing them in to `set` so that we're guaranteed `key?` will exist.

@tubbo, would you mind cutting a new gem with this fix in it? :)

Just for others searching, here's the actual error message:

`NoMethodError undefined method 'key?' for #<ActionDispatch::Request::Session::Options>`

It took me a while to find this issue :D

**Comment by Lianowar** [comment]
@brendon thank you to provide actual message:) I hope that search engines catch and index this:)

**Comment by edmondchui** [comment]
Thanks @brendon for the explanation. I just update redis-rack to version 2.1.3 and the issue is resolved.

**Comment by tubbo** [comment]
@edmondchui @Lianowar @brendon omg, i'm so sorry for not pinging you folks here yesterday! knew i was forgetting something...anyway thank you very much for the PR and help ensuring this issue doesn't affect anyone else down the road. :) much appreciated.

**Comment by edmondchui** [comment]
Thanks @tubbo. I was lucky to check yesterday and found out you released version 2.1.3 12 hours before I checked.
